### PR TITLE
FAL-1751 Configure mailing list software

### PR DESCRIPTION
**Pull request by swalladge**
[FAL-1751](https://tasks.opencraft.com/browse/FAL-1751)

Add ansible scripts for configuring a mailing list.

Work in progress.

**Test instructions**:

- Set configuration variables
- Run this against an instance
- Verify that mailman3, mailman3-web, nginx proxy, postfix are all set up.

**Reviewers**:

- [x] @toxinu

### CODE REVIEW

File: playbooks/roles/mailing-list/templates/postfix-main.cf [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# Only relay mail from the local mailman server
+mynetworks_style = host
+
+# Support the default VERP delimiter.
+recipient_delimiter = +
+
+# Don't treat owner- and -request addresses specially
+owner_request_special = no
+
+# Let mailman3 manage the transport and relay
```
Comment (swalladge) [inline playbooks/roles/mailing-list/templates/postfix-main.cf:23]: @tartansandal I thought postfix was supposed to do the transport part?

Comment (tartansandal) [reply playbooks/roles/mailing-list/templates/postfix-main.cf:23]: >  thought postfix was supposed to do the transport part?

@swalladge Yup that is exactly what we are doing.  What the above setting does is allow `mailman` to manage how `postfix` handles the transport (by changing its configuration). This way you can add/modify mailing lists in mailman without having to explicitly touch the postfix transport configuration. Make sense?

File: playbooks/roles/mailing-list/handlers/main.yml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+---
+- name: "restart mailman3"
+  service:
+    name: "{{ item }}"
+    state: restarted
```
Comment (tartansandal) [inline playbooks/roles/mailing-list/handlers/main.yml:5]: This should really be two separate handlers.

File: playbooks/roles/mailing-list/handlers/main.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  with_items:
+    - "mailman3"
+    - "mailman3-web"
```
Comment (tartansandal) [inline playbooks/roles/mailing-list/handlers/main.yml:8]: @swalladge This should really be two separate handlers (`notify` takes a list). And you don't need all the quotes.

File: playbooks/roles/mailing-list/templates/postfix-main.cf (hunk 1 above)
Comment (swalladge) [reply playbooks/roles/mailing-list/templates/postfix-main.cf]: Yep, cool :+1:

**Review by toxinu — APPROVED** [review]
:+1:

- [x] I tested this: just browse the new playbook and role
- [x] I read through the code

**Pull request merged**